### PR TITLE
TASK: Restrict typo3fluid/fluid to versions before signature change

### PR DESCRIPTION
**Pull request by kdambekalns**
See https://github.com/neos/flow-development-collection/issues/2260

### CODE REVIEW

File: Neos.FluidAdaptor/composer.json [hunk 1]
```diff
@@ -8,7 +8,7 @@
 		"neos/cache": "*",
 		"neos/utility-files": "*",
 		"neos/utility-objecthandling": "*",
-		"typo3fluid/fluid": ">=2.1.3,<2.5.0"
+		"typo3fluid/fluid": "~2.1.3 < 2.1.7 || >=2.2.0 < 2.2.4 || ~2.3.0 < 2.3.7 || ~2.4.0 <2.4.4"
```
Comment (albe) [inline Neos.FluidAdaptor/composer.json:11]:
```suggestion
		"typo3fluid/fluid": "~2.1.3 < 2.1.7 || ~2.2.0 < 2.2.4 || ~2.3.0 < 2.3.7 || ~2.4.0 < 2.4.4"
```

File: composer.json [hunk 2]
```diff
@@ -22,7 +22,7 @@
         "symfony/dom-crawler": "~2.8.0",
         "symfony/console": "~2.8",
         "neos/composer-plugin": "^2.0.0",
-        "typo3fluid/fluid": ">=2.1.3,<2.5.0",
+        "typo3fluid/fluid": "~2.1.3 < 2.1.7 || >=2.2.0 < 2.2.4 || ~2.3.0 < 2.3.7 || ~2.4.0 <2.4.4",
```
Comment (albe) [inline composer.json:25]:
```suggestion
        "typo3fluid/fluid": "~2.1.3 < 2.1.7 || ~2.2.0 < 2.2.4 || ~2.3.0 < 2.3.7 || ~2.4.0 < 2.4.4",
```

**Review by albe — APPROVED** [review]
Just have a fetish for consistency 🙈

**Comment by kdambekalns** [comment]
> Just have a fetish for consistency 🙈

Thanks! I changed that "last minute"–fully agree!

File: composer.json [hunk 3]
```diff
@@ -22,7 +22,7 @@
         "symfony/dom-crawler": "~2.8.0",
         "symfony/console": "~2.8",
         "neos/composer-plugin": "^2.0.0",
-        "typo3fluid/fluid": ">=2.1.3,<2.5.0",
+        "typo3fluid/fluid": "~2.1.3 <2.1.7 || ~2.2.0 <2.2.4 || ~2.3.0 <2.3.7 || ~2.4.0 <2.4.4",
```
Comment (bwaidelich) [inline composer.json:25]: what have they done to us? o.O

**Comment by bwaidelich** [comment]
Thanks for taking care!
According to https://semver.mwl.be/#!?package=typo3fluid%2Ffluid&version=~2.1.3%20%3C2.1.7%20%7C%7C%20~2.2.0%20%3C2.2.4%20%7C%7C%20~2.3.0%20%3C2.3.7%20%7C%7C%20~2.4.0%20%3C2.4.4&minimum-stability=stable this would disallow version 2.5.x but according to #2260 only 2.5.11+ is problematic. Was that done on purpose?

**Comment by bwaidelich** [comment]
> Was that done on purpose?

ah, 4.3 only allowed <2.5.0 before.. So I guess it _was_ :)

**Comment by bwaidelich** [comment]
This change was applied to `master` (with https://github.com/neos/fluidadaptor/commit/a997367797b1c18b8f5f6e9301a9954b11e3246a). Was that intended?

**Comment by albe** [comment]
Wait - https://github.com/neos/flow-development-collection/blob/master/Neos.FluidAdaptor/composer.json#L14 ?

**Comment by kdambekalns** [comment]
That commit (https://github.com/neos/fluidadaptor/commit/a997367797b1c18b8f5f6e9301a9954b11e3246a) isn't listed on https://github.com/neos/fluidadaptor/commits/master/composer.json - what you see is this:
Screenshot 2020-12-03 at 22 06 48

*Yes*, that commit is in master. But *while upmerging* it was "discarded", as I took care to not merge wrong things. So it ended up correctly in master.

**Comment by bwaidelich** [comment]
Ah right, thanks for clarifying!

**Pull request merged**